src/state/profile/index.tsx

```diff
@@ -12,21 +12,24 @@ const initialState: ProfileState = {
   profileAvatars: {},
 }
 
-export const fetchProfileAvatar = createAsyncThunk<{ account: string; nft: Nft }, string>(
+export const fetchProfileAvatar = createAsyncThunk<{ account: string; nft: Nft; hasRegistered: boolean }, string>(
   'profile/fetchProfileAvatar',
   async (account) => {
-    const nft = await getProfileAvatar(account)
-    return { account, nft }
+    const { nft, hasRegistered } = await getProfileAvatar(account)
+    return { account, nft, hasRegistered }
   },
 )
 
-export const fetchProfileUsername = createAsyncThunk<{ account: string; username: string }, string>(
-  'profile/fetchProfileUsername',
-  async (account) => {
-    const username = await getUsername(account)
-    return { account, username }
-  },
-)
+export const fetchProfileUsername = createAsyncThunk<
+  { account: string; username: string },
+  { account: string; hasRegistered: boolean }
+>('profile/fetchProfileUsername', async ({ account, hasRegistered }) => {
+  if (!hasRegistered) {
+    return { account, username: '' }
```

Comment:
Return empty username and don't make requests for username. Needed to set the fetching flag to FETCHED